flybywiresim/a32nx
# feat(efb): flypad weight and balance/payload loading page v1 🚀

Fixes #6254
Fixes #3297
Fixes #7362

## Summary of Changes

- Added option to load passengers and cargo in the EFB
  - Supports total passenger and Cargo entry
  - Supports modification of specific passenger station weights by clicking on the seating layout
- Added ZFW entry (outputs an appropriate passenger/cargo load)
- Added per passenger weight/per bag weight entry
- Added Per Seat View
- Added Per Cargo Location Entry (click to set)
- Added Dynamic CG vs Payload Chart
- Added on-page boarding time modification (similar to fuel page)
- Updated A20N SVGs (credit to LeDerpLegend#0109)

## Screenshots (if necessary)


![image](https://user-images.githubusercontent.com/15316958/181244101-0984248a-ad1a-4328-98d5-2c03d1cdaa02.png)


![image](https://user-images.githubusercontent.com/15316958/181244726-846dde4f-4058-4036-a687-3f6ac4e73871.png)







https://user-images.githubusercontent.com/15316958/179195256-bd54c20b-547e-4c7f-99ce-8dac2302259c.mp4


https://user-images.githubusercontent.com/15316958/179195381-20583ef8-8922-480b-b883-6bb531f0166b.mp4

https://user-images.githubusercontent.com/15316958/179195313-d1fefa38-836b-43af-84cd-cb59a1a0d4de.mp4


## References

![image](https://user-images.githubusercontent.com/15316958/179195001-a6f4bd7f-6e7b-4f00-a41a-1c8d91c9cf6e.png)

https://www.airbus.com/sites/g/files/jlcbta136/files/2021-11/Airbus-Commercial-Aircraft-AC-A320.pdf





## Additional context



Discord username (if different from GitHub):2Cas#1022

## Testing instructions

1. Go to Ground > Services page and check that all elements are correctly rendered
2. Go to Payload page, 
3. Adjust seating by clicking on the seating layout. This should add new seats and increase the table value for planned passengers.
4. Add baggage by clicking on the 4 sliders, this should affect the desired cargo value in the table.
5. Change per passenger and per bag weight from 80 and 15 to higher/lower values (or leave at default)
6. Input desired passenger and desired cargo in the table
7. Check that ZFW calculation is accurate (Empty Weight is 42500KG, add passenger, passenger bags and cargo weight to this value and this should match desired ZFW)
8. Board passengers with desired time setting (Real/Fast/Instant). Should be able to change speed while passengers are boarding and have it transition between boarding times seamlessly.
9. Deboard passengers (either by inputting in the table, or clicking on the seating layout).
10. Go to the main EFB dashboard by clicking the FBW Logo in the top left, import a simbrief OFP
11. Go back to the payload page and click the cloud icon next to the table to load the OFP. Check that all relevant table values are correctly imported.
12. Load passengers/cargo again
13. Fill out INIT-A and INIT-B in MCDU (Init fuel pred). Check that MLDW indicator now shows on the performance envelope chart.
14. Proceed with normal flight operations. Check that when in air, MTOW envelope is replaced by the FLIGHT envelope.
15. Check that visuals in all themes (dark/light/blue) are correct and readable, with both bright daytime and at night (in sim time)

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
